src/catch2/catch_message.cpp

```diff
@@ -106,10 +107,20 @@ namespace Catch {
     }
 
     void Capturer::captureValue( size_t index, std::string const& value ) {
-        assert( index < m_messages.size() );
-        m_messages[index].message += value;
-        m_resultCapture.pushScopedMessage( m_messages[index] );
-        m_captured++;
+        if(isScoped){
+            assert( index < m_messages.size() );
+            m_messages[index].message += value;
+            m_resultCapture.pushScopedMessage( m_messages[index] );
+            m_captured++;
+        } else {
+            getResultCapture().emplaceUnscopedMessage(Catch::MessageBuilder(
+                m_messages[index].macroName,
+                m_messages[index].lineInfo,
+                m_messages[index].type) << value);
+            if(index == m_messages.size() - 1){
+                m_messages.clear();
+            }
```

Comment:
No need to clear this here, instead check in destructor whether `m_isScoped` is true and change it accordingly.